test/e2e/workflow/deployment_test.go

```diff
@@ -454,6 +455,124 @@ var _ = Describe("NetworkAddonsConfig", func() {
+		var (
+			configSpec            cnao.NetworkAddonsConfigSpec
+			ovsDaemonSet          = &v1.DaemonSet{}
+			kubemacpoolDeployment = &v1.Deployment{}
+		)
+
+		BeforeEach(func() {
+			configSpec = cnao.NetworkAddonsConfigSpec{
+				Ovs:         &cnao.Ovs{},
+				KubeMacPool: &cnao.KubeMacPool{},
+			}
+
+			CreateConfig(gvk, configSpec)
+			CheckConfigCondition(gvk, ConditionAvailable, ConditionTrue, 15*time.Minute, CheckDoNotRepeat)
+		})
+
+		getDaemonsetLabels := func(ds *v1.DaemonSet) map[string]string { return ds.GetLabels() }
+		getDaemonsetTemplateLabels := func(ds *v1.DaemonSet) map[string]string { return ds.Spec.Template.GetLabels() }
+		getDeploymentLabels := func(ds *v1.Deployment) map[string]string { return ds.GetLabels() }
+		getDeploymentTemplateLabels := func(ds *v1.Deployment) map[string]string { return ds.Spec.Template.GetLabels() }
+
+		Context("When NetworkAddonsConfig is labeled with `part-of`, `version` and `component` labels", func() {
+			const (
+				partOfLabelCRVal     = "part-of-cr-value"
+				versionLabelCRVal    = "version-cr-value"
+				componentLabelCRlVal = "component-cr-value"
+			)
+			labels := map[string]string{names.PART_OF_LABEL_KEY: partOfLabelCRVal,
+				names.VERSION_LABEL_KEY: versionLabelCRVal, names.COMPONENT_LABEL_KEY: componentLabelCRlVal}
+
+			BeforeEach(func() {
+				LabelConfig(gvk, labels)
+
+				Eventually(func() map[string]string {
+					return GetConfig(gvk).GetLabels()
+				}, 2*time.Minute, time.Minute).Should(SatisfyAll(
+					HaveKeyWithValue(names.PART_OF_LABEL_KEY, partOfLabelCRVal),
+					HaveKeyWithValue(names.VERSION_LABEL_KEY, versionLabelCRVal),
+					HaveKeyWithValue(names.COMPONENT_LABEL_KEY, componentLabelCRlVal)))
+			})
+
+			It("Should take the `part-of`, `version` and `component` labels value and apply it to the deployed daemonset", func() {
+				Eventually(func() *v1.DaemonSet {
+					err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: OvsComponent.DaemonSets[0], Namespace: components.Namespace}, ovsDaemonSet)
+					Expect(err).ToNot(HaveOccurred(), "should succeed getting the ovs daemonSet")
+					return ovsDaemonSet
+				}, 2*time.Minute, time.Minute).Should(SatisfyAll(
+					WithTransform(getDaemonsetLabels, HaveKeyWithValue(names.PART_OF_LABEL_KEY, partOfLabelCRVal)),
+					WithTransform(getDaemonsetLabels, HaveKeyWithValue(names.VERSION_LABEL_KEY, versionLabelCRVal)),
+					WithTransform(getDaemonsetLabels, HaveKeyWithValue(names.COMPONENT_LABEL_KEY, componentLabelCRlVal)),
+					WithTransform(getDaemonsetTemplateLabels, HaveKeyWithValue(names.PART_OF_LABEL_KEY, partOfLabelCRVal)),
+					WithTransform(getDaemonsetTemplateLabels, HaveKeyWithValue(names.VERSION_LABEL_KEY, versionLabelCRVal)),
+					WithTransform(getDaemonsetTemplateLabels, HaveKeyWithValue(names.COMPONENT_LABEL_KEY, componentLabelCRlVal))), "label and template label values of the daemonset should be contain the values on NetworkAddonsConfig")
```

Comment:
please consider making this a function , and then this and the next test can be both in one `It` 
that calls this function with parameters (either deployment / ds)

Reply:
I would preferably use a function and send it the labels then do `Expect(getDaemonsetLabels).To(ContainElements(expectedLabels)` 
but that is not supported as I want to compare both key+Value and:
```
For maps, ContainElements matches against the map’s values.
so UUIC I can only check it like that...
```